File: tools/io_throttle.py
#!/usr/bin/env python
import sys, re, time

limit = int(sys.argv[1])

def getTokens():
    return int(re.search("io_tokens=([0-9]+)",
        open("/proc/io_status").read()).group(1))

while 1:
    line = sys.stdin.readline()
    if not line:
        break
    while getTokens() < limit:
        time.sleep(0.1)
    sys.stdout.write(line)

File: tools/io_throttle.pl
#!/usr/bin/env perl

$limit = $ARGV[1];

while (<STDIN>) {
    print;
    while (1) {
        open STATUS, "/proc/io_status";
        <STATUS> =~ /io_tokens=([0-9]+)/;
        if ($1 < $limit) {
            sleep(0.5);
        }
        else {
            break;
        }
    }
}
